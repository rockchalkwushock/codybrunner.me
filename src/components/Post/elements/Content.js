import styled from 'styled-components'

/**
 * @fileOverview
 *
 * This is the styling for all html generated
 * from markdown.
 *
 * themeing is based on: theme.post.<key>
 */

const Content = styled.section`
  padding-top: 1em;

  > h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${({ theme }) => theme.post.ftFm};
    margin: 0.75em 0;
  }

  > ol,
  p,
  ul {
    color: ${({ theme }) => theme.post.ftClr};
    font-family: ${({ theme }) => theme.post.ftFm};
    font-size: ${({ theme }) => theme.post.ftSz};
    font-weight: ${({ theme }) => theme.post.ftWt};
    line-height: ${({ theme }) => theme.post.lnHt};
    padding: 0.5em 0;
  }

  > ol,
  ul {
    margin-left: 1.5em;
  }

  > ol li,
  ul li {
    margin: 0.25em 0;
  }

  > blockquote p em a,
  > ol li a,
  > p a,
  > ul li a {
    color: ${({ theme }) => theme.post.linkClr};
    font-weight: ${({ theme }) => theme.post.linkWt};
    text-decoration: underline ${({ theme }) => theme.post.linkClr};
    transition: all 0.2s ease-in-out;
    :visited {
      color: ${({ theme }) => theme.post.linkClr};
    }
    :hover {
      color: ${({ theme }) => theme.post.linkHvClr};
    }
  }

  > blockquote {
    background-color: rgba(0, 0, 0, 0.1);
    border-left: 5px solid ${({ theme }) => theme.site.bg};
    border-radius: 5px;
    margin: 0.5em auto;
    padding: 0.25em 0.75em;
    width: 100%;
  }

  > blockquote p {
    color: rgba(0, 0, 0, 0.8);
    font-style: italic;
    font-size: 1.1rem;
  }

  > ul li p {
    padding: 0;
  }

  > ul .task-list-item p {
    color: blue;
  }

  > blockquote:last-child {
    background-color: #000;
  }

  > blockquote p:last-child {
    text-align: center;
  }

  > p:last-child {
    text-align: center;
  }

  .gatsby-resp-image-link {
    padding: 0.25em 0;
  }

  .gatsby-highlight pre {
    background-color: rgba(0, 0, 0, 0.1);
  }

  @media (min-width: ${({ theme }) => theme.screen.large}) {
    > blockquote {
      width: 62%;
    }
  }
`

Content.displayName = 'Content'

export default Content
