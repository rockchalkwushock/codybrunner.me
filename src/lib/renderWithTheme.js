import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
