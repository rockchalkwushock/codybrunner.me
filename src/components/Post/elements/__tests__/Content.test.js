import React from 'react'

import renderWithTheme from '../../../../lib/renderWithTheme'
import Content from '../Content'

const content = 'hello world'

// TODO: Better testing here
// I need a real content object with all the elements to test their styles.

describe('Component: <Content />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(
      <Content dangerouslySetInnerHTML={{ __html: content }} />
    )
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('padding-top', '1em')
  })
})
