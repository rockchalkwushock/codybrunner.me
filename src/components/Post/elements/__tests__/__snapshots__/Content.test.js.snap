// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Content /> should render correctly 1`] = `
.c0 {
  padding-top: 1em;
}

.c0 > h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-family: "Raleway",sans-serif;
  margin: 0.75em 0;
}

.c0 > ol,
.c0 p,
.c0 ul {
  color: #000;
  font-family: "Raleway",sans-serif;
  font-size: 1.1rem;
  font-weight: 400;
  line-height: 1.618;
  padding: 0.5em 0;
}

.c0 > ol,
.c0 ul {
  margin-left: 1.5em;
}

.c0 > ol li,
.c0 ul li {
  margin: 0.25em 0;
}

.c0 > blockquote p em a,
.c0 > ol li a,
.c0 > p a,
.c0 > ul li a {
  color: #143a52;
  font-weight: 600;
  text-decoration: underline #143a52;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0 > blockquote p em a:visited,
.c0 > ol li a:visited,
.c0 > p a:visited,
.c0 > ul li a:visited {
  color: #143a52;
}

.c0 > blockquote p em a:hover,
.c0 > ol li a:hover,
.c0 > p a:hover,
.c0 > ul li a:hover {
  color: #7a9eb1;
}

.c0 > blockquote {
  background-color: rgba(0,0,0,0.1);
  border-left: 5px solid #7a9eb1;
  border-radius: 5px;
  margin: 0.5em auto;
  padding: 0.25em 0.75em;
  width: 100%;
}

.c0 > blockquote p {
  color: rgba(0,0,0,0.8);
  font-style: italic;
  font-size: 1.1rem;
}

.c0 > ul li p {
  padding: 0;
}

.c0 > ul .task-list-item p {
  color: blue;
}

.c0 > blockquote:last-child {
  background-color: #000;
}

.c0 > blockquote p:last-child {
  text-align: center;
}

.c0 > p:last-child {
  text-align: center;
}

.c0 .gatsby-resp-image-link {
  padding: 0.25em 0;
}

.c0 .gatsby-highlight pre {
  background-color: rgba(0,0,0,0.1);
}

@media (min-width:736px) {
  .c0 > blockquote {
    width: 62%;
  }
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Content
    dangerouslySetInnerHTML={
      Object {
        "__html": "hello world",
      }
    }
  >
    <section
      className="c0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "hello world",
        }
      }
    />
  </Content>
</ThemeProvider>
`;
